recipe/meta.yaml

```diff
@@ -1,46 +1,79 @@
 {% set name = "splink" %}
-{% set version = "3.6.0" %}
+{% set version = "3.9.10" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 6bbef574b12ebf74e112960e2bc5d8baf3ca60f5d889a581dafb2f29173e90a8
+  sha256: 891996d3eaa6ff58cc3e3cd97c0aa0c45246a8b1c858cd4fe77f617e190f20ee
 
 build:
-  noarch: python
-  number: 1
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<38 or s390x]
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
+# the requirements refer to the duckdb-less installation documented at
+# https://moj-analytical-services.github.io/splink/getting_started.html#duckdb-less-installation
+# https://github.com/moj-analytical-services/splink/blob/v3.9.10/scripts/duckdbless_requirements.txt
 requirements:
   host:
-    - python >=3.7,<4.0
+    - python
     - pip
-    - poetry >=0.12
+    - poetry-core 1.5.2
   run:
-    - python >=3.7,<4.0
-    - jsonschema >=3.2,<5.0
-    - python-duckdb >=0.6.0,<0.7.0
-    - pandas >=1.0.0,<2.0.0
-    - sqlglot >=5.1.0
-    - altair >=4.2.0
-    - jinja2 >=3.0.3,<4.0.0
-  run_constrained:
-    - pyspark >=3.2.1
+    - python
+    - altair >=5.0.1,<6.0.0
+    - attrs >=23.1.0
+    - greenlet >=2.0.2
+    - importlib_resources >=5.4.0 # [py<39]
+    - jinja2 >=3.1.2
+    - jsonschema >=4.17.3,<5.0
+    - markupsafe >=2.1.3
+    - pandas >=1.3.5
+    - phonetics >=1.0.5,<2.0.0
+    - pkgutil-resolve-name >=1.3.10  #[py<=38]
+    - psycopg2 >=2.9.7
+    - py4j >=0.10.9.7
+    - pyrsistent >=0.19.3
+    - pyspark >=3.4.1
+    - python-dateutil >=2.8.2
+    - pytz >=2023.3.post1
+    - six >=1.16.0
+    - sqlalchemy >=1.4.49,<2.0.0
+    - sqlglot >=13.0.0,<19.0.0
+    - toolz >=0.12.0
+    - typing-extensions >=4.7.1
+    - zipp >=3.15.0
 
 test:
   imports:
     - splink
+  commands:
+  # while doing pip check got an error: splink 3.9.10 requires duckdb, which is not installed.
```

Comment:
if this fails this way, the package dependencies are pretty much broken... should be fixed rather than skipped

Reply:
if the dependency is optional, as upstream claims (I can see at least one place out of the `splink.duckdb` package where it is imported, it should be marked as optional in `pyproject.toml` via a patch, instead of the documented workaround.

Reply:
what are you suggesting to do in this case? changing the project toml, like so?

> [tool.poetry.dependencies]
> duckdb = {version=">=0.8.0", optional=true}

even then, should I add all dependencies documented in [duckdbless_requirements.txt](https://github.com/moj-analytical-services/splink/blob/v3.9.10/scripts/duckdbless_requirements.txt) to the `[tool.poetry.dependencies]` section of the `pyproject.toml`?

Reply:
`pip check` will fail anyway because `duckdb` is hardcoded in `pyproject.toml` https://inspector.pypi.io/project/splink/3.9.10/packages/e1/e8/9f15de9bd9e3279a43c712e554037a409e0c606471682ed4a7baab47e640/splink-3.9.10.tar.gz/splink-3.9.10/pyproject.toml#line.16. 
`duckdb` is imported only in one place https://github.com/moj-analytical-services/splink/blob/ac1cf23d41215492b68d56841c46de3bd93c7e99/splink/comparison_helpers.py#L1. I think it's not related to the _duckdbless_ variant.
I would say that removing `duckdb` is an upstream workaround https://github.com/moj-analytical-services/splink/blob/ac1cf23d41215492b68d56841c46de3bd93c7e99/scripts/duckdb-less_packages.sh#L14

If we want to maintain this recipe with `duckdb` in the future we can move `duckdb` as a dependency to the optional group of dependencies. If we don't plan to maintain it we can ignore `pip check`'s warnings and use the current patch or modify it to make duckdb as optional